pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/TypeDefinition.java

```diff
@@ -4,6 +4,10 @@
 
 package net.sourceforge.pmd.lang.java.typeresolution.typedefinition;
 
+import net.sourceforge.pmd.annotation.InternalApi;
+
+@Deprecated
+@InternalApi
 public interface TypeDefinition {
```

Comment:
Is this really internal? Or will the API be different in PMD 7.0.0?
Well, the interface doesn't offer much functionality anyway...

Reply:
Yeah I just don't know what that interface is good for. All the codebase uses JavaTypeDefinition instead